test/integration/etcd/server.go

```diff
@@ -152,19 +138,24 @@ func StartRealAPIServerOrDie(t *testing.T, configFuncs ...func(*options.ServerRu
 
 	kubeClient := clientset.NewForConfigOrDie(kubeClientConfig)
 
+	stopCh := make(chan struct{})
+	errCh := make(chan error)
 	go func() {
 		// Catch panics that occur in this go routine so we get a comprehensible failure
 		defer func() {
 			if err := recover(); err != nil {
 				t.Errorf("Unexpected panic trying to start API server: %#v", err)
 			}
 		}()
+		defer close(errCh)
 
 		prepared, err := kubeAPIServer.PrepareRun()
 		if err != nil {
-			t.Error(err)
+			errCh <- err
+			return
 		}
 		if err := prepared.Run(stopCh); err != nil {
+			errCh <- err
 			t.Error(err)
```

Comment:
return?

Reply:
It's the last thing in this function anyway. 
But it wouldn't hurt either - added.

Reply:
lol, sorry, you replaced the t.Error(err) by a return in the previous line, since the err is now processed in the other side of the channel.
I was asking if you wanted to replace this t.Error(err) by a return too, but keep it this way, it doesn't hurt